apm-sniffer/apm-sdk-plugin/httpasyncclient-4.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/httpasyncclient/v4/LeaseRequestCompletedInterceptor.java

```diff
@@ -33,7 +33,7 @@ public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allAr
             MethodInterceptResult result) throws Throwable {
         PoolEntry entry = (PoolEntry) allArguments[0];
         NHttpConnection conn = (NHttpConnection) entry.getConnection();
-        if (ContextManager.isActive()) {
+        if (ContextManager.isActive() && !ContextManager.activeSpan().isExit()) {
```

Comment:
I think this doesn't have real impact with or without this check. Could you explain more?

Reply:
This will cause the behavior of the first request to be inconsistent with that of the second

Reply:
The inconsistent usually means a bug in real case. Could you be more clear? Because adding more checks would impact the runtime performance, we should add them unless bugs happen.